### PR TITLE
Write a .mailmap file

### PR DESCRIPTION
**Pull request by LukeShu**
## Description

I was trying to make some charts about number of contributors, and was thrown off by the same person showing up multiple ways; so I started writing a .mailmap file to normalize idents.

 - For outside contributors who have used more than 1 non-`@users.noreply.github.com` email address, I put zero effort in to determining which address they prefer; I just picked one arbitrarily.

 - I've gone ahead and moved myself to "Former employees"--I'm only here for another week.

This is the command I use to visually scan for idents that might be duplicates:

    git log --all --format=$'%cN <%cE>\n%aN <%aE>' | sort | uniq -c

## Related Issues

(none)

## Testing

I've spent a lot of time looking at that above `git log` command looking for names that might be duplicates.

## Checklist

- [ ] **Does my change need to be backported to a previous release?** - no
  - What backport versions were discussed with the Maintainers in the Issue?

- [ ] **I made sure to update `CHANGELOG.md`.** - no applicable changes

   Remember, the CHANGELOG needs to mention:
  - Any new features
  - Any changes to our included version of Envoy
  - Any non-backward-compatible changes
  - Any deprecations

- [ ] **This is unlikely to impact how Ambassador performs at scale.** - not a runtime change

   Remember, things that might have an impact at scale include:
  - Any significant changes in memory use that might require adjusting the memory limits
  - Any significant changes in CPU use that might require adjusting the CPU limits
  - Anything that might change how many replicas users should use
  - Changes that impact data-plane latency/scalability

- [ ] **My change is adequately tested.**

   Remember when considering testing:
  - Your change needs to be specifically covered by tests.
    - Tests need to cover all the states where your change is relevant: for example, if you add a behavior that can be enabled or disabled, you'll need tests that cover the enabled case and tests that cover the disabled case. It's not sufficient just to test with the behavior enabled.
  - You also need to make sure that the _entire area being changed_ has adequate test coverage.
    - If existing tests don't actually cover the entire area being changed, add tests.
    - This applies even for aspects of the area that you're not changing – check the test coverage, and improve it if needed!
  - We should lean on the bulk of code being covered by unit tests, but...
  - ... an end-to-end test should cover the integration points

- [ ] **I updated `DEVELOPING.md` with any any special dev tricks I had to use to work on this code efficiently.**

- [ ] **The changes in this PR have been reviewed for security concerns and adherence to security best practices.**

### CODE REVIEW

File: .mailmap [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+Nick Powell <nickpowell@datawire.io>                   <nickjaypowell@gmail.com>
+Rafael Schloming <rhs@datawire.io>                     <rhs@alum.mit.edu>
+Thomas Hallgren <thomas@datawire.io>                   <thomas@tada.se>
+William Hardin <williamhardin@datawire.io>             <47009048+w-h37@users.noreply.github.com>
+
+# Former Ambassador Labs employees (without a new address)
+Alix Cook <alixcook@datawire.io>                       <73843761+acookin@users.noreply.github.com>
+Douglas Camata <dcamata@datawire.io>                   <159076+douglascamata@users.noreply.github.com>
+Leonardo Luz Almeida <leonardoluz@datawire.io>         <leoluz@users.noreply.github.com>
+Matt McClure <mattmcclure@datawire.io>                 <73489204+mattmcclure-dw@users.noreply.github.com>
```
Comment (kflynn) [inline .mailmap:57]: So... we don't have an updated way to reach these people? Is it worth trying to find one?

File: .mailmap [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+Philip Lombardi <plombardi89@gmail.com>                <plombardi@datawire.io>
+Shahriar Rostami <srdsecond@gmail.com>                 <54422255+alphashr@users.noreply.github.com>
+Stephanie Coyle <stephaniecoyle391@gmail.com>
+Stephanie Coyle <stephaniecoyle391@gmail.com>          <55929576+scoyle391@users.noreply.github.com>
+Stephanie Coyle <stephaniecoyle391@gmail.com>          <scoyle@datawire.io>
+
+# Outside contributors (inconsistent email -- did I choosed the correct email?)
+Brandon Catcho <bcatcho@doubledutch.me>                <bcatcho@gmail.com>
+Christoph Grossegger <christoph.grossegger@gmail.com>  <christoph@dust-bit-games.com>
+Dan Sipple <dsipple@etsy.com>                          <sipple.dan@gmail.com>
```
Comment (kflynn) [inline .mailmap:98]: Dan Sipple was at Datawire for awhile, actually, possibly as a contractor? Not sure it matters much.

File: .mailmap [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+Endre Czirbesz <endre.czirbesz@rungway.com>            <endre@czirbesz.hu>
+Endre Czirbesz <endre.czirbesz@rungway.com>            <endrec@users.noreply.github.com>
+Giovanni Gargiulo <gargiulo.gianni@gmail.com>          <giovanni.gargiulo@hbc.com>
+John Morrissey <jmorrissey@devoted.com>                <jwm@devoted.com>
+Josue Diaz <josue_diaz@affirmednetworks.com>           <josuesdiaz15@msn.com>
+Markus Maga <markus@maga.nu>                           <markus@maga.se>
```
Comment (kflynn) [inline .mailmap:104]: I always thought of `maga.se` as the canonical address for Markus, but I guess we could ask.

File: .mailmap [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+Niko Kurtti <niko.kurtti@interdax.com>                 <niko@kurtti.eu>
+Phil Peble <philpeble@gmail.com>                       <ppeble@activecampaign.com>
```
Comment (kflynn) [inline .mailmap:106]: I think that Phil always contributed as an ActiveCampaign employee. Might be worth checking, he's easy to reach on Slack.

File: .mailmap [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+Wenhua Zhao <whzhao@gmail.com>                         <wenhuaz@splunk.com>
+n1koo <niko.kurtti@interdax.com>                       <niko@kurtti@eu>
+
+# Outside contributors (inconsistent name)
+Kevin Dagostino <kevin@tonkworks.com>
+Kowalczyk Bartek <bkowalczyyk@gmail.com>
+Mohit Sharma <imoisharma@icloud.com>
+
+# Outside contributors (just @users.norely.github.com substitutions)
+Jordan Neufeld <jordan.neufeld@nfl.com>                <neufeldtech@users.noreply.github.com>
```
Comment (kflynn) [inline .mailmap:117]: Also probably worth checking: I thought Nuefeld Tech tried to be careful about the distinction between them and the NFL? Might be misremembering, of course.

**Review by LanceEa — DISMISSED** [review]

File: .mailmap (hunk 1 above)
Comment (LukeShu) [reply .mailmap:57]: I put a little bit of effort in, these are the few folks that I didn't find one for. I guess the next step would be reaching out on LinkedIn?

File: .mailmap (hunk 2 above)
Comment (LukeShu) [reply .mailmap:98]: He was a contractor, and never had an `@datawire.io` email. The "employee/former-employee/non-employee" distinctions were for me, answering "should `@datawire.io` be canonical".

File: .mailmap (hunk 5 above)
Comment (LukeShu) [reply .mailmap:117]: The `@nfl.com` email is the only one he's ever committed with other than the `@users.noreply.github.com`.

File: .mailmap (hunk 1 above)
Comment (LukeShu) [reply .mailmap:57]: And to be clear: I didn't look for former-employees in general; this is just the ones that used multiple emails to commit with.

File: .mailmap (hunk 4 above)
Comment (LukeShu) [reply .mailmap:106]: OK, I've sent 12 emails of the form

```
Date: Tue, 15 Nov 2022 10:14:33 -0700
Message-ID: <emissary-mailmap-1@lukeshu.com>
From: Luke Shumaker <lukeshu@lukeshu.com>
To: Brandon Catcho <bcatcho@doubledutch.me>, Brandon Catcho <bcatcho@gmail.com>
Subject: Attribution for Emissary/Ambassador contributions
User-Agent: Wanderlust/2.15.9 (Almost Unreal) SEMI-EPG/1.14.7 (Harue)
 FLIM-LB/1.14.9 (Gojō) APEL-LB/10.8 EasyPG/1.0.0
 Emacs/28.2 (x86_64-pc-linux-gnu) MULE/6.0 (HANACHIRUSATO)
MIME-Version: 1.0 (generated by SEMI-EPG 1.14.7 - "Harue")
Content-Type: text/plain; charset=US-ASCII

Hi Brandon,

I'm with Emissary-ingress (formerly Ambassador API Gateway), and am
setting up a Git `.mailmap` file to normalize contributor
information.  I see that there are multiple email addresses that
you've committed with; which address would you prefer to be the
canonical/current one?

  - bcatcho@doubledutch.me
  - bcatcho@gmail.com

-- 
Cheers,
~ Luke Shumaker
```
for these 12 sets of names/addresses
```
Brandon Catcho <bcatcho@doubledutch.me>                <bcatcho@gmail.com>
Christoph Grossegger <christoph.grossegger@gmail.com>  <christoph@dust-bit-games.com>
Dan Sipple <dsipple@etsy.com>                          <sipple.dan@gmail.com>
Endre Czirbesz <endre.czirbesz@rungway.com>            <endre@czirbesz.hu> 
Giovanni Gargiulo <gargiulo.gianni@gmail.com>          <giovanni.gargiulo@hbc.com>
John Morrissey <jmorrissey@devoted.com>                <jwm@devoted.com>
Josue Diaz <josue_diaz@affirmednetworks.com>           <josuesdiaz15@msn.com>
Markus Maga <markus@maga.nu>                           <markus@maga.se>
Niko Kurtti <niko.kurtti@interdax.com>                 <niko@kurtti.eu>
Phil Peble <philpeble@gmail.com>                       <ppeble@activecampaign.com>
Romario Maxwell <maxwellra@jncb.com>                   <romario.maxwell@gmail.com>
Wenhua Zhao <whzhao@gmail.com>                         <wenhuaz@splunk.com>
```

File: .mailmap (hunk 1 above)
Comment (LukeShu) [reply .mailmap:57]: OK, looking through the other `@datawire.io` emails: other former-employees without new addresses are Bruce Horn, David Dymko, and Ryan Park.

Comment (LukeShu) [reply .mailmap:57]: OK, I've reached out to 5 of those 6 people on LinkedIn... I couldn't find Ryan. Or rather, I found a whole bunch of Ryan Parks but couldn't quickly tell which one is him.

**Comment by LukeShu** [comment]
I've pushed 4 new commits revising it further.

File: .mailmap (hunk 2 above)
Comment (LukeShu) [reply .mailmap]: I've now added a comment about that.

**Review by LanceEa — APPROVED** [review]

**Review by kflynn — APPROVED** [review]
Thanks! Hit it.

**Pull request merged**